src/test/java/uk/gov/hmcts/darts/audio/service/impl/AudioTransformationServiceImplTest.java

```diff
@@ -142,4 +156,77 @@ void testHandleKedaInvocationForMediaRequestsCaseNull() {
         assertThrows(NullPointerException.class, () -> audioTransformationService.handleKedaInvocationForMediaRequests());
 
     }
+
+    @Test
+    void filterMediaByMediaRequestDatesWithStartDateExactRequestAndEndDateExactRequest() {
+        List<MediaEntity> mediaEntities = createMediaEntities(TIME_12_00, TIME_12_20, TIME_12_20, TIME_12_40, TIME_12_40, TIME_13_00);
+        MediaRequestEntity mediaRequestEntity = createMediaRequest(TIME_12_00, TIME_13_00);
+        List<MediaEntity> mediaEntitiesResult = audioTransformationService.filterMediaByMediaRequestDates(mediaEntities, mediaRequestEntity);
+
+        assertEquals(3, mediaEntitiesResult.size());
+        assertEquals(TIME_12_00, mediaEntitiesResult.get(0).getStart());
+        assertEquals(TIME_13_00, mediaEntitiesResult.get(2).getEnd());
+    }
+
+    @Test
+    void filterMediaByMediaRequestDatesWithStartDateAfterRequestAndEndDateBeforeRequest() {
+        List<MediaEntity> mediaEntities = createMediaEntities(TIME_12_01, TIME_12_20, TIME_12_20, TIME_12_40, TIME_12_40, TIME_12_59);
+        MediaRequestEntity mediaRequestEntity = createMediaRequest(TIME_12_00, TIME_13_00);
+        List<MediaEntity> mediaEntitiesResult = audioTransformationService.filterMediaByMediaRequestDates(mediaEntities, mediaRequestEntity);
+        assertEquals(3, mediaEntitiesResult.size());
+        assertEquals(TIME_12_01, mediaEntitiesResult.get(0).getStart());
+        assertEquals(TIME_12_59, mediaEntitiesResult.get(2).getEnd());
+    }
+
+    @Test
+    void filterMediaByMediaRequestDatesWithStartDateBeforeRequestAndEndDateAfterRequest() {
+        List<MediaEntity> mediaEntities = createMediaEntities(TIME_11_59, TIME_12_20, TIME_12_20, TIME_12_40, TIME_12_40, TIME_13_01);
+        MediaRequestEntity mediaRequestEntity = createMediaRequest(TIME_12_00, TIME_13_00);
+        List<MediaEntity> mediaEntitiesResult = audioTransformationService.filterMediaByMediaRequestDates(mediaEntities, mediaRequestEntity);
+        assertEquals(2, mediaEntitiesResult.size());
```

Comment:
Can you check if `2` is correct? Since the media request timespan is entirely contained within the overall timespan of all 3 medias, shouldn't this be `3`?

Reply:
There would be nothing to trim of 12:20 as mentioned above